src/earn/EarnAddCrypto.tsx

```diff
@@ -0,0 +1,162 @@
+    >
+      <View style={styles.actionsContainer}>
+        {actions.map((action) => (
+          <Touchable
+            style={styles.touchable}
+            key={action.name}
+            borderRadius={20}
+            onPress={() => {
+              ValoraAnalytics.track(EarnEvents.earn_tap_add_crypto, {
+                action: action.name,
+                ...getTokenAnalyticsProps(token),
+              })
+              action.onPress()
+            }}
+            testID={`Earn/AddCrypto/${action.name}`}
+          >
+            <>
+              <action.iconComponent color={Colors.black} />
+              <View style={{ flex: 1 }}>
+                <Text style={styles.actionTitle}>{action.title}</Text>
+                <Text style={styles.actionDetails}>{action.details}</Text>
+              </View>
+            </>
+          </Touchable>
+        ))}
+      </View>
+    </BottomSheet>
+  )
+}
+
+export const getActions = (token: TokenBalance, tokenAmount: BigNumber) => {
+  const { t } = useTranslation()
+  const { swappableFromTokens } = useSwappableTokens()
+  const cashInTokens = useCashInTokens()
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+
+  const addAmount = {
+    crypto: tokenAmount.toNumber(),
+    fiat: Math.round(
+      (useTokenToLocalAmount(tokenAmount, token.tokenId) || new BigNumber(0)).toNumber()
+    ),
+  }
+
+  return [
+    {
+      name: TokenActionName.Add,
+      title: t('earnFlow.addCrypto.actions.add'),
+      details: t('earnFlow.addCrypto.actionDescriptions.add', {
+        tokenSymbol: token.symbol,
+        tokenNetwork: NETWORK_NAMES[token.networkId],
+      }),
+      iconComponent: QuickActionsAdd,
+      onPress: () => {
+        navigate(Screens.SelectProvider, {
+          tokenId: token.tokenId,
+          flow: CICOFlow.CashIn,
+          amount: addAmount,
+        })
+      },
+      visible: !!cashInTokens.find((tokenInfo) => tokenInfo.tokenId === token.tokenId),
+    },
+    {
+      name: TokenActionName.Receive,
+      title: t('earnFlow.addCrypto.actions.receive'),
+      details: t('earnFlow.addCrypto.actionDescriptions.receive', {
+        tokenSymbol: token.symbol,
+        tokenNetwork: NETWORK_NAMES[token.networkId],
+      }),
+      iconComponent: QuickActionsSend,
+      onPress: () => {
+        navigate(Screens.ExchangeQR, { flow: CICOFlow.CashIn, exchanges: [] })
+      },
+      visible: true,
+    },
+    {
+      name: TokenActionName.Swap,
+      title: t('earnFlow.addCrypto.actions.swap'),
+      details: t('earnFlow.addCrypto.actionDescriptions.swap', {
+        tokenSymbol: token.symbol,
+        tokenNetwork: NETWORK_NAMES[token.networkId],
+      }),
+      iconComponent: QuickActionsSwap,
+      onPress: () => {
+        navigate(Screens.SwapScreenWithBack, { toTokenId: token.tokenId })
+      },
+      visible:
+        isSwapEnabled &&
+        !!swappableFromTokens.find((tokenInfo) => tokenInfo.networkId === token.networkId),
```

Comment:
this should compare tokenId's right?

Reply:
I don't think so, swap should be enabled if we have a token that we can swap into USDC on arbitrum, so we want to check if there is a swappable token on the Arbitrum network, so I am comparing network IDs.

I guess I should also check that there is a match where the token ID is not USDC since we won't swap that for itself.